### PR TITLE
feat(VETS-CPC-837): add pre-defined review for a rating to make user feedback more flexible and friendly

### PR DESCRIPTION
**Pull request by Mila5847**
JIRA: link to jira ticket
CPC-837
https://champlainsaintlambert.atlassian.net/jira/software/c/projects/CPC/boards/13?selectedIssue=CPC-837&sprints=313
## Context:
This ticket is about including a pre-defined review for when a user add or update a rating of a specific vet. Previously, the user was only able to write a description (explanation) about the rate score they gave for the vet. Now, they will be able to select it as well, if they want to. This makes it easier for the users who want to give written feedback but do not/cannot type it themselves.
## Changes
- Change made in Rating entity, Rating Response DTO, Rating Request DTO to include "private PredefinedDescription predefinedDescription"
- Change made in addRatingToVet method in the vets-service to include the possibility of a selected pre-defined review description.
- Change made in updateRatingByVetIdAndRatingId method in the vets-service to include the possibility of a selected pre-defined review description.
- Change made in vet-details.controller.js and vet-details.template.html to include the predefinedDescription and its display.
- Added tests in VetsServiceClientIntegrationTest and ApiGatewayControllerIntegration test, as well as in RatingRepositoryTest and VetControllerIntegrationTest
- Changed methods (add rating and update rating) are tested > 90%.
## Before and After UI (Required for UI-impacting PRs)
Before
image
After
image
image
## Dev notes (Optional)
- Specific technical changes that should be noted
## Linked pull requests (Optional)
- pull request link

### CODE REVIEW

File: api-gateway/src/main/resources/static/scripts/vet-details/vet-details.template.html [hunk 1]
```diff
@@ -99,128 +103,135 @@ <h3 class="form-label-rating text-center" style="margin-top: 5%">Educations</h3>
...
-                                    <a class="btn btn-danger educationButton" href="javascript:void(0)" ng-click="deleteVetEducation(educations.educationId)">Delete Education</a>
-                                    </td>
-                            </tr>
+                        <tr ng-repeat="educations in $ctrl.educations track by educations.educatonId">
+                            <td>{{ educations.degree }}</td>
+                            <td>{{ educations.schoolName }}</td>
+                            <td>{{ educations.fieldOfStudy }}</td>
+                            <td>{{ educations.startDate }}</td>
+                            <td>{{ educations.endDate }}</td>
+                        </tr>
```
Comment (Sprytte) [inline api-gateway/src/main/resources/static/scripts/vet-details/vet-details.template.html:115]:
```suggestion
                        <tr ng-repeat="educations in $ctrl.educations track by educations.educatonId">
                            <td>{{ educations.degree }}</td>
                            <td>{{ educations.schoolName }}</td>
                            <td>{{ educations.fieldOfStudy }}</td>
                            <td>{{ educations.startDate }}</td>
                            <td>{{ educations.endDate }}</td>
                            <td>
                                    <a class="btn btn-danger educationButton" href="javascript:void(0)" ng-click="deleteVetEducation(educations.educationId)">Delete Education</a>
                                    </td>
                            </tr>
```
Did you accidentally delete (hah) the delete button?

**Review by Sprytte — APPROVED** [review]
Overall looks good! Code is well implemented both in the front-end and back-end and the tests were appropriately modified

File: api-gateway/src/main/resources/static/scripts/vet-details/vet-details.template.html (hunk 1 above)
Comment (Mila5847) [reply api-gateway/src/main/resources/static/scripts/vet-details/vet-details.template.html:115]: I sure did, sorry!! Fixing it right now

**Review by DenisaNicoletaH — APPROVED** [review]
The provided code looks well done. It contains tests that are well done and the naming conventions are also good. In other words, this feature is nicely done

**Review by Sprytte — APPROVED** [review]
Nice to see the button back!

**Pull request merged**